src/wazuh_modules/wm_database.c

```diff
@@ -264,15 +264,17 @@ void wm_sync_manager() {
                 // Get os_major
                 if (w_regexec("^([0-9]+)\\.*", os_version, 2, match)) {
                     match_size = match[1].rm_eo - match[1].rm_so;
-                    os_major = malloc(match_size +1);
-                    snprintf (os_major, match_size +1, "%.*s", match_size, os_version + match[1].rm_so);
+
+                    if (os_major = malloc(match_size +1), os_major)
+                        snprintf (os_major, match_size +1, "%.*s", match_size, os_version + match[1].rm_so);
                 }
 
                 // Get os_minor
                 if (w_regexec("^[0-9]+\\.([0-9]+)\\.*", os_version, 2, match)) {
                     match_size = match[1].rm_eo - match[1].rm_so;
-                    os_minor = malloc(match_size +1);
-                    snprintf (os_minor, match_size +1, "%.*s", match_size, os_version + match[1].rm_so);
+
+                    if (os_minor = malloc(match_size +1), os_minor)
```

Comment:
Here we can do the same that @palaciosjeremias suggests about **os_malloc()**

Reply:
DONE